### PR TITLE
[SPARK-36506][PYTHON] Improve test coverage for series.py and indexes/*.py.

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes improving test coverage for pandas-on-Spark Series & Index code base, which is written in `series.py` and `indexes/*.py` separately.

This PR did the following to improve coverage:
- Add unittest for untested code
- Fix unittest which is not tested properly
- Remove unused code

**NOTE**: This PR is not only include the test-only update, for example it includes the new warning for `__xor__`, `__and__`, `__or__`.

### Why are the changes needed?

To make the project healthier by improving coverage.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Unittest.

### CODE REVIEW

File: python/pyspark/pandas/indexes/base.py [hunk 1]
```diff
@@ -2601,6 +2599,12 @@ def __iter__(self) -> Iterator:
         return MissingPandasLikeIndex.__iter__(self)
 
     def __xor__(self, other: "Index") -> "Index":
+        warnings.warn(
+            "Index.__xor__ operating as a set operation is deprecated, "
+            "in the future this will be a logical operation matching Series.__xor__.  "
+            "Use index.symmetric_difference(other) instead",
+            FutureWarning,
```
[the remark this answers is not in the document]
Comment (itholic) [reply python/pyspark/pandas/indexes/base.py:2606]: added!

**Comment by itholic** [comment]
rebased!

**Review by ueshin — APPROVED** [review]
LGTM.

File: python/pyspark/pandas/indexes/base.py [hunk 2]
```diff
@@ -2601,6 +2599,12 @@ def __iter__(self) -> Iterator:
         return MissingPandasLikeIndex.__iter__(self)
 
     def __xor__(self, other: "Index") -> "Index":
+        warnings.warn(
+            "Index.__xor__ operating as a set operation is deprecated, "
+            "in the future this will be a logical operation matching Series.__xor__.  "
```
[the remark this answers is not in the document]
Comment (itholic) [reply python/pyspark/pandas/indexes/base.py:2604]: Oh, yeah.

Let me take a look soon. Thanks!

[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply python/pyspark/pandas/indexes/base.py:2604]: @itholic is there any update on this?

[the remark this answers is not in the document]
Comment (itholic) [reply python/pyspark/pandas/indexes/base.py:2604]: SPARK-36653 is working in progress at https://github.com/apache/spark/pull/33911.

Let me review.

**Review by xinrong-meng — APPROVED** [review]

**Comment by xinrong-databricks** [comment]
Thanks @itholic !

File: python/pyspark/pandas/indexes/base.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ueshin) [reply python/pyspark/pandas/indexes/base.py:2604]: Btw, pandas `Index` has `__and__` and `__or__` mapping to `intersection` and `union` respectively.
They are already deprecated, but show the similar messages.
Although they will be removed soon, they might be good to have to show the messages which might be helpful to users.
WDYT? cc @HyukjinKwon

[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply python/pyspark/pandas/indexes/base.py:2604]: Yeah, let's do that.

**Comment by HyukjinKwon** [comment]
@itholic mind resolving conflicts?

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
@itholic mind updating Pr description too?

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**